addons/binding/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoAlbumArt.java

```diff
@@ -0,0 +1,122 @@
+package org.openhab.binding.onkyo.internal;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class to handle Onkyo Album Arts.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public class OnkyoAlbumArt {
+
+    private Logger logger = LoggerFactory.getLogger(OnkyoAlbumArt.class);
+
+    private enum State {
+        INVALID,
+        NOTSTARTED,
+        STARTED,
+        NEXT,
+        READY;
+    }
+
+    private StringBuilder albumArtStringBuilder = new StringBuilder();
+    private String imageType = "unknown";
+    private State state = State.NOTSTARTED;
+
+    public String getImageType() {
+        return imageType;
+    }
+
+    public boolean isAlbumCoverTransferStarted() {
+        return state == State.STARTED;
+    }
+
+    public boolean isAlbumCoverReady() {
+        return state == State.READY;
+    }
+
+    public void clearAlbumArt() {
+        albumArtStringBuilder.setLength(0);
+        imageType = "unknown";
+        state = State.NOTSTARTED;
+    }
+
+    public void addFrame(String data) {
+
+        if (data.length() <= 2) {
+            return;
+        }
+
+        char imgType = data.charAt(0);
+
+        switch (imgType) {
```

Comment:
In case of my receiver, I've always received album art with type=2
(from the spec: t-> Image type 0:BMP, 1:JPEG, 2:URL, n:No Image)

the best handling would be to load the pic from the url and present it here as well.
Similar was recently introduced in the sonos binding as well https://github.com/eclipse/smarthome/pull/2838